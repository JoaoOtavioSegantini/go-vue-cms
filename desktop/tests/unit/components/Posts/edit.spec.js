import { mount } from "@vue/test-utils";
import EditPage from "@/components/Posts/Edit.vue";
import ClassicEditor from "@ckeditor/ckeditor5-build-classic";
import { createStore } from "vuex";

describe("EditPost.vue", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  window.axios = {
    get: jest
      .fn(() => {
        return {
          then: jest.fn(() => "your faked response"),
        };
      })
      .mockName("axiosget"),
    put: jest
      .fn(() => {
        return {
          then: jest.fn(() => "your faked response"),
        };
      })
      .mockName("axiosPut"),
  };

  const mockRoute = {
    params: {
      id: 1,
    },
  };
  const mockRouter = {
    push: jest.fn(),
  };
  const createVuexStore = () => {
    return createStore({
      state() {
        return {
          Posts: {
            onePost: {
              ID: "1",
              title: "Post 1",
              body: "A simple post text body",
              slug: "simple-slug-text",
            },
          },
        };
      },
      mutations: {
        updatePost(state) {
          state.onePost = {
            ID: "1",
            slug: "John Doe",
            title: "Post 1",
            body: "A simple post text body",
          };
        },
      },
      actions: {
        getPost(ctx, res) {
          ctx.commit("updatePost", res);
        },
        updatePost(ctx, data) {},
      },
    });
  };

  function factory() {
    const store = createVuexStore();

    return mount(EditPage, {
      mocks: {
        axios: window.axios,
      },
      global: {
        plugins: [store],
        mocks: {
          $router: mockRouter,
          $route: mockRoute,
        },
      },
      data() {
        return {
          editor: ClassicEditor,
        };
      },
    });
  }

  it("renders props.msg when passed", async () => {
    const msg = "AÇÕES Post 1 Edição de artigo do blogTítuloURLConteúdoSalvar";
    const wrapper = factory();

    wrapper.findAll("input")[0].setValue("acbdfg@gmail.com");
    wrapper.findAll("input")[1].setValue("987654");

    await wrapper.find("form").trigger("submit");
    await wrapper.find("button").trigger("click");
    await wrapper.vm.save();
    expect(mockRouter.push).toBeCalledWith({ path: "/posts/1" });
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
