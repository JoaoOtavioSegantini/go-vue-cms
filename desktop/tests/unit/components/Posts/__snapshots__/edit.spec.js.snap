// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPost.vue renders props.msg when passed 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col-1 nav\\">
      <p class=\\"display-6\\"><small>AÇÕES</small></p><a href=\\"/posts/1\\" class=\\"btn btn-secondary\\" title=\\"Voltar\\"><i class=\\"fa fa-arrow-circle-left\\"></i></a>
    </div>
    <div class=\\"col-11\\">
      <div class=\\"card\\">
        <div class=\\"card-body\\">
          <h5 class=\\"title\\"><i class=\\"fa fa-file-text-o\\"></i> Post 1 <small>Edição de artigo do blog</small></h5>
          <form>
            <div class=\\"form-group\\"><label for=\\"title\\">Título</label><input type=\\"text\\" class=\\"form-control\\" id=\\"title\\" placeholder=\\"Título da página\\"></div>
            <div class=\\"form-group\\"><label for=\\"slug\\">URL</label><input type=\\"text\\" class=\\"form-control\\" id=\\"slug\\" placeholder=\\"Url da página\\"></div>
            <div class=\\"form-group\\"><label for=\\"body\\">Conteúdo</label>
              <ckeditor editor=\\"class nC extends Ng{}\\" modelvalue=\\"A simple post text body\\"></ckeditor>
            </div><button type=\\"submit\\" class=\\"btn btn-primary\\">Salvar</button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
